components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/APIDefinition.java

```diff
@@ -205,4 +205,13 @@ public abstract String processOtherSchemeScopes(String resourceConfigsJSON)
     public abstract API setExtensionsToAPI(String swaggerContent, API api)
             throws APIManagementException;
 
+    /**
+     * This method will extract scopes from legacy x-wso2-security and add them to default scheme
+     * @param swaggerContent swagger definition
+     * @return
+     * @throws APIManagementException
+     */
+    public abstract String processLegacyScopes(String swaggerContent)
```

Comment:
I think we should not expose processLegacyScopes via the interface because I think it is not required to execute directly. 
I think we can define same method in respective parsers as private and call inside the processOtherSchemeScopes method. Because processLegacyScopes is used by only processOtherSchemeScopes.

Reply:
fixed with https://github.com/wso2/carbon-apimgt/pull/9012/commits/b188ce5404a9c6d4634d3c3ad7d44eba254e4437